### PR TITLE
chore: force delete addon manager when addon manager pods get stuck on terminating state

### PR DESCRIPTION
**Pull request by JesusAlvarezTorres**
**Reason for Change**:

Workaround an issue seen on 1.15 to 1.16 upgrade on Azure Stack disconnected scenario where the CSE will timeout on deleting the addon manager pods; the pods go into terminating state and are not deleted unless force delete is invoked.

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -395,7 +395,7 @@ ensureAddons() {
   wait_for_file 1200 1 $ADDON_MANAGER_SPEC || exit {{GetCSEErrorCode "ERR_FILE_WATCH_TIMEOUT"}}
   sed -i "s|${ADDONS_DIR}/init|${ADDONS_DIR}|g" $ADDON_MANAGER_SPEC || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
   {{/* Force re-load all addons because we have changed the source location for addon specs */}}
-  retrycmd 120 5 30 ${KUBECTL} delete pods -l app=kube-addon-manager -n kube-system || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
+  retrycmd 120 5 30 ${KUBECTL} delete pods -l app=kube-addon-manager -n kube-system --force --grace-period 0 || exit {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}
```
Comment (mboersma) [inline parts/k8s/cloud-init/artifacts/cse_config.sh:398]: Should we first try a normal (polite) `kubectl delete pods` before adding `--force -grace-period 0`?

Comment (JesusAlvarezTorres) [reply parts/k8s/cloud-init/artifacts/cse_config.sh]: change made

**Comment by jackfrancis** [comment]
@JesusAlvarezTorres did we root cause *why* the pod delete operations are stalling indefinitely?

**Comment by JesusAlvarezTorres** [comment]
> @JesusAlvarezTorres did we root cause _why_ the pod delete operations are stalling indefinitely?

@jackfrancis no, we didn't. The issue seems to be intermittent, while validating this change we noticed on some clusters the graceful delete works while others had to fall back to the force delete, will need some time to see why this is happening

**Comment by jackfrancis** [comment]
Also @JesusAlvarezTorres can you change the title of the PR, this change doesn't actually care about 1.15--1.16, it's enabling the --force short-circuit foo for every CSE run (create/scale/upgrade).

**Comment by JesusAlvarezTorres** [comment]
> Also @JesusAlvarezTorres can you change the title of the PR, this change doesn't actually care about 1.15--1.16, it's enabling the --force short-circuit foo for every CSE run (create/scale/upgrade).

@jackfrancis Sure, updated title.

**Pull request merged**